mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -404,7 +410,10 @@ void CMomentumPlayer::Spawn()
     SetBodygroup(1, 11); // BODY_PROLATE_ELLIPSE
     // BASECLASS SPAWN MUST BE AFTER SETTING THE MODEL, OTHERWISE A NULL HAPPENS!
     BaseClass::Spawn();
-    AddFlag(FL_GODMODE);
+    
+    static ConVarRef gm("mom_gamemode");
+    if (gm.GetInt() != GAMEMODE_RJ)
+        AddFlag(FL_GODMODE);
```

Comment:
Now I worry about this, because there are elements that could damage a player found in maps.

Something to confirm for me: does the player take fall damage on this game mode? Can it kill them?

Reply:
Further thought: could enforcing godmode on, but only checking it when it's *not* a rocket/stickybomb explosion giving damage work? May require implementing the base class stuff but tweaking its code to do so.

Reply:
I would assume this blocks all damage except damage from rockets, but maybe falldamage or map entities ignore this?: https://github.com/momentum-mod/game/blob/9899f59624614340708277f8ae1427bea917cac0/mp/src/game/shared/momentum/mom_gamerules.cpp#L350-L357
Most speedrun servers in TF2 just set soldier's health to some high value every tick to prevent dying. Speedrun servers also remove any flinching effects, so in rocketjumping not taking any damage should be fine.  
However, I think taking damage will be a necessity for sticky jumping, otherwise you could just plop 8 sticks down and not die which isn't really the intended mechanic.

Reply:
I have an idea. Try setting `m_takedamage = DAMAGE_EVENTS_ONLY` as most things [check it](https://github.com/momentum-mod/game/blob/e3efc48eb58cf56af61b2fef8d0bf7e3bf754680/mp/src/game/server/baseentity.cpp#L1437) before doing things with health.

Reply:
Been pretty busy this week, will take a look at this during the weekend.

Reply:
`m_takedamage = DAMAGE_EVENTS_ONLY` seems to work. I couldn't find any effects from falling, explosions, etc. and it allows rocketjumping without removing godmode.